File: palestra-transform-otim-obj-kotlin/src/main/java/transformacao/java/dtoscomplexos/objetos/Venda.java
package transformacao.java.dtoscomplexos.objetos;

import java.math.BigDecimal;
import java.time.LocalDateTime;
import java.util.List;

public class Venda {
    private LocalDateTime dataVenda;
    private Cliente cliente;
    private BigDecimal valorTotal;
    private List<VendaItem> itensVenda;

    public LocalDateTime getDataVenda() {
        return dataVenda;
    }

    public void setDataVenda(LocalDateTime dataVenda) {
        this.dataVenda = dataVenda;
    }

    public Cliente getCliente() {
        return cliente;
    }

    public void setCliente(Cliente cliente) {
        this.cliente = cliente;
    }

    public BigDecimal getValorTotal() {
        return valorTotal;
    }

    public void setValorTotal(BigDecimal valorTotal) {
        this.valorTotal = valorTotal;
    }

    public List<VendaItem> getItensVenda() {
        return itensVenda;
    }

    public void setItensVenda(List<VendaItem> itensVenda) {
        this.itensVenda = itensVenda;
    }
}

File: palestra-transform-otim-obj-kotlin/src/main/kotlin/transformacao/kotlin/dtoscomplexos/beans/Venda.kt
package transformacao.kotlin.dtoscomplexos.beans

import java.math.BigDecimal
import java.time.LocalDateTime

data class Venda(
        val numero: Int,
        val cliente: Cliente,
        val valorTotal: BigDecimal,
        val dataVenda: LocalDateTime,
        val itens: List<VendaItem>)

data class VendaItem(
        val quantidade: BigDecimal,
        val valorUnitario: BigDecimal,
        val valorTotal: BigDecimal,
        val produto: Produto
)